### PR TITLE
Set registration buffer to three months; refactor to Result object

### PR DESCRIPTION
**Pull request by rsanheim**
**Story card:** https://app.clubhouse.io/simpledotorg/story/651/update-patient-registration-queries

### CODE REVIEW

File: spec/services/control_rate_service_spec.rb [hunk 1]
```diff
@@ -143,6 +147,28 @@ def refresh_views
     expect(result[:uncontrolled_patients_rate][Period.month(june_1_2020)]).to eq(50.0)
   end
 
+  it "excludes patients registeed in the last 3 months" do
```
Comment (rsanheim) [inline spec/services/control_rate_service_spec.rb:150]: 👍

File: spec/services/no_bp_measure_service_spec.rb [hunk 2]
```diff
@@ -5,51 +5,55 @@
...
 
-  let(:june_1_2018) { Time.parse("June 1, 2018 00:00:00+00:00") }
-  let(:june_1_2020) { Time.parse("June 1, 2020 00:00:00+00:00") }
-  let(:june_30_2020) { Time.parse("June 30, 2020 00:00:00+00:00") }
-  let(:july_2020) { Time.parse("July 15, 2020 00:00:00+00:00") }
-  let(:jan_2019) { Time.parse("January 1st, 2019 00:00:00+00:00") }
-  let(:jan_2020) { Time.parse("January 1st, 2020 00:00:00+00:00") }
-  let(:july_2018) { Time.parse("July 1st, 2018 00:00:00+00:00") }
-  let(:july_2020) { Time.parse("July 1st, 2020 00:00:00+00:00") }
-
```
Comment (rsanheim) [inline spec/services/no_bp_measure_service_spec.rb:16]: @timcheadle any reason you got rid of these? i believe Hari added the timezone info to try to catch TZ boundary condition issues.

Comment (timcheadle) [reply spec/services/no_bp_measure_service_spec.rb:16]: They weren't being used anywhere?

Comment (rsanheim) [reply spec/services/no_bp_measure_service_spec.rb:16]: Oh.  thats a good reason =)  I thought this was a different spec, my mistake.

**Pull request merged**